src/client.rs

```diff
@@ -269,38 +269,43 @@ impl NamedPipeClientReader {
 
     /// Read full message/bytes into a vec
     pub fn read_full(&self) -> Result<Vec<u8>, Error> {
-        let buffer_size = self.options.read_buffer_size;
-
-        let mut buffer: Vec<u8> = Vec::with_capacity(buffer_size as usize);
-
-        let mut buffer_to_read = buffer_size;
-        let mut buffer_ptr = 0;
-        let mut read_bytes = 0;
-
-        while unsafe {
-            ReadFileSys(
-                self.handle.0 .0,
-                buffer.as_mut_ptr().add(buffer_ptr) as *mut _,
-                buffer_to_read,
-                &mut read_bytes,
-                std::ptr::null_mut(),
-            ) == 0
-        } {
+        let buffer_size = std::cmp::max(self.options.read_buffer_size as usize, 1);
```

Comment:
This is a good catch for the buffer size, however this check is better placed in [`read_buffer_size()`](https://github.com/MolotovCherry/WinPipes-rs/blob/8e7331fac0eff8a53e8e60098119aaa31d613d00/src/client.rs#L193) instead as a sanity check, since it should already be assumed the set read buffer size is valid by the time it reaches this.

We can just do a `assert!(size > 0, "size must be > 0");` there since the size should never be 0